scripts/cc-cli/src/utils/account.ts

```diff
@@ -14,69 +15,105 @@ export function initECDSAKeyringPairFromPK(pk: string) {
     console.error(getErrorMessage(e));
     process.exit(1);
   }
 }
-export async function getCallerSeedFromEnvOrPrompt(interactive: boolean) {
+
+export async function initCallerKeyring(
+  options: OptionValues,
+): Promise<KeyringPair> {
   try {
-    return await getSeedFromEnvOrPrompt("CC_SEED", "caller", interactive);
+    return await initKeyringFromEnvOrPrompt("CC_SECRET", "caller", options);
   } catch (e) {
     console.error(getErrorMessage(e));
     process.exit(1);
   }
 }
 
-async function getSeedFromEnvOrPrompt(
-  envVar = "CC_SEED",
-  accountRole = "caller",
-  interactive = true,
-) {
+export async function initKeyringFromEnvOrPrompt(
+  envVar: string,
+  accountRole: string,
+  options: OptionValues,
+): Promise<KeyringPair> {
+  const interactive = options.input;
+  const ecdsa = options.ecdsa;
+  const inputName = ecdsa ? "private key" : "seed phrase";
+  const validateInput = ecdsa ? validateECDSAKey : mnemonicValidate;
+  const generateKeyring = ecdsa ? initECDSAKeyringPairFromPK : initKeyringPair;
+
   if (!interactive && !process.env[envVar]) {
     throw new Error(
-      `Error: Must specify a seed phrase for the ${accountRole} account in the environment variable ${envVar} or use an interactive shell.`,
+      `Error: Must specify a ${inputName} for the ${accountRole} account in the environment variable ${envVar} or use an interactive shell.`,
     );
   }
 
   if (typeof process.env[envVar] === "string") {
-    const seedFromEnv = process.env[envVar];
-    if (mnemonicValidate(seedFromEnv!)) {
-      return seedFromEnv;
+    const input = getStringFromEnvVar(process.env[envVar]);
+    if (validateInput(input)) {
+      return generateKeyring(input);
     } else {
       throw new Error(
         `Error: Seed phrase provided in environment variable ${envVar} is invalid.`,
       );
     }
   } else if (interactive) {
-    const seedPromptResult = await prompts([
+    const promptResult = await prompts([
       {
         type: "password",
         name: "seed",
-        message: `Specify a seed phrase for the ${accountRole} account`,
-        validate: (seed) => mnemonicValidate(seed),
+        message: `Specify a ${inputName} for the ${accountRole} account`,
+        validate: (input) => validateInput(input),
       },
     ]);
     // If SIGTERM is issued while prompting, it will log a bogus address anyways and exit without error.
     // To avoid this, we check if prompt was successful, before returning.
-    if (seedPromptResult.seed) {
-      return seedPromptResult.seed;
+    if (promptResult.seed) {
+      return generateKeyring(promptResult.seed);
     }
   }
-  throw new Error("Error: Could not retrieve seed phrase.");
+  throw new Error(`Error: Could not retrieve ${inputName}`);
+}
+
+export function validateECDSAKey(pk: string): boolean {
+  const keyring = initECDSAKeyringPairFromPK(pk);
+  const msg = "";
+  const sig = keyring.sign(msg);
+  return keyring.verify(msg, sig, keyring.publicKey);
+}
```

Comment:
I don't see how this can ever return `false`. Any given string is a valid seed for a keypair, so `initECDSAKeyringPairFromPK` will always return a valid keypair (just potentially with an unexpected private key if you typoed the PK). Then this function is effectively testing that a valid keypair creates a valid signature, which will always be the case (unless polkadotJS is broken).

Reply:
Yes, previous validation was just `() => true`. All strings are valid ECDSA PKs, so there is no check. We can go back to `true` in the future if we want to prevent wasting resources on creating a keyring and simulating a signature.